File: ios/protocols/RNFCMProvider.swift
import Foundation
import Altcraft

@objcMembers
final class RNFCMProvider: NSObject, FCMInterface {
  private let get: (@escaping (String?) -> Void) -> Void
  private let del: (@escaping (Bool) -> Void) -> Void

  init(
    get: @escaping (@escaping (String?) -> Void) -> Void,
    del: @escaping (@escaping (Bool) -> Void) -> Void
  ) {
    self.get = get
    self.del = del
  }

  func getToken(completion: @escaping (String?) -> Void) {
    get(completion)
  }

  func deleteToken(completion: @escaping (Bool) -> Void) {
    del(completion)
  }
}

File: android/src/main/java/com/altcraft/sdk/interfaces/RnFCMProvider.kt
package com.altcraft.sdk.rn.providers

import androidx.annotation.Keep
import com.altcraft.sdk.interfaces.FCMInterface


@Keep
object RnFCMProvider : FCMInterface {
    @Volatile
    private var token: String? = null

    override suspend fun getToken(): String? = token

    override suspend fun deleteToken(completion: (Boolean) -> Unit) {
        token = null
        completion(true)
    }

    fun setToken(newToken: String?) {
        token = newToken
    }

    fun clear() {
        token = null
    }
}